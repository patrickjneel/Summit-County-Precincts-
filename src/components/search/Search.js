import React, { Component } from 'react';
import APIKey from '../../apiKey/ApiKey';
import './search.css';

class Search extends Component {
  constructor() {
    super();
    this.state = {
      search: ''
    }
  }
  
  handleInput = (event) => {
    this.setState({
      search: event.target.value
    })
  }

  // findLocation = async () => {
  //   const fetchedLocation = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?address=${this.state.search}&key=${APIKey}`)
  //   const locationData = await fetchedLocation.json();
  //   const mappedLocation = locationData.results.map(place => {
  //     console.log(place)
  //     console.log(place.address_components[1].short_name)
  //     console.log(place.geometry.location)
  //     debugger;
  //     return place.address_components[1].short_name && place.geometry.location

      // this needs to be a conected compoent 
      // send the address to the store 
      // make a some sort of card list component 
      // that has access to your data and the filter string in the store 
      // filter through the data by the search adddress 
      // once you have that single address object check to see if the address is within the address range || included in the full address array. 

      // if thats true that is your precinct 
  //   })
  //   this.setState({search: ''})

  // }

  render() {
    return (
    <div className="search-area">
      <h2 className="title">Precinct-Pro</h2>
      <input 
          className="search-bar"
          value = {this.state.search}
          onChange={this.handleInput}
          placeholder="Search for Address"/>
      <button onClick={this.findLocation}>
      Search
      </button>
    </div>
  )
  }
}

export default Search;
