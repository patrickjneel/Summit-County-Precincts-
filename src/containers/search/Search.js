import React, { Component } from 'react';
import { connect } from 'react-redux';
import api from '../../API/API';
import APIKey from '../../apiKey/ApiKey';
import { addLocation } from '../../actions/Actions';
import './search.css';

export class Search extends Component {
  constructor() {
    super();
    this.state = {
      search: ''
    }
  }
  
  handleInput = (event) => {
    this.setState({
      search: event.target.value
    })
  }

   findLocation = async () => {
    const fetchedLocation = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?address=${this.state.search}&key=${APIKey}`)
    const locationData = await fetchedLocation.json();
    const mappedLocation = locationData.results.map(place => {
  
      let coordinates =  {
        address: place.address_components[1].short_name,
        coordinates: place.geometry.location
      }
      console.log(coordinates)
      this.props.addLocation(coordinates)
    })
  }
      // send the address to the store 
      // make a some sort of card list component 
      // that has access to your data and the filter string in the store 
      // filter through the data by the search adddress 
      // once you have that single address object check to see if the address is within the address range || included in the full address array. 

      // if thats true that is your precinct 
   


  render() {
    return (
    <div className="search-area">
      <h2 className="title">Precinct-Pro</h2>
      <input 
          className="search-bar"
          value = {this.state.search}
          onChange={this.handleInput}
          placeholder="Search for Address"/>
      <button onClick={this.findLocation}>
      Search
      </button>
    </div>
  )
  }
}

export const mapStateToProps = (store) => {
  return {
    coords: store.coords
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
    addLocation: (coords) => {
      dispatch(addLocation(coords))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Search);
