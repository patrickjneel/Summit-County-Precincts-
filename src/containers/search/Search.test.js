import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search Tests', () => {

  it('should be defined', () => {
  const renderedSearch = shallow(<Search />);

  expect(renderedSearch).toBeDefined();
  });

  it('should match the snapshot', () => {
    const renderedSearch = shallow(<Search />);

    expect(renderedSearch).toMatchSnapshot();
  });
})
