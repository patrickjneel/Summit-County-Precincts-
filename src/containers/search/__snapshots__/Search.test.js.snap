// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
>
          Precinct-Pro
</h2>,
        <div
          className="search-bottom"
>
          <input
                    className="search-bar"
                    onChange={[Function]}
                    placeholder="Search for Address"
                    value=""
          />
          <button
                    onClick={[Function]}
          >
                    Search
          </button>
</div>,
      ],
      "className": "search-area",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Precinct-Pro",
          "className": "title",
        },
        "ref": null,
        "rendered": "Precinct-Pro",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="search-bar"
              onChange={[Function]}
              placeholder="Search for Address"
              value=""
/>,
            <button
              onClick={[Function]}
>
              Search
</button>,
          ],
          "className": "search-bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-bar",
              "onChange": [Function],
              "placeholder": "Search for Address",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
>
            Precinct-Pro
</h2>,
          <div
            className="search-bottom"
>
            <input
                        className="search-bar"
                        onChange={[Function]}
                        placeholder="Search for Address"
                        value=""
            />
            <button
                        onClick={[Function]}
            >
                        Search
            </button>
</div>,
        ],
        "className": "search-area",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Precinct-Pro",
            "className": "title",
          },
          "ref": null,
          "rendered": "Precinct-Pro",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="search-bar"
                onChange={[Function]}
                placeholder="Search for Address"
                value=""
/>,
              <button
                onClick={[Function]}
>
                Search
</button>,
            ],
            "className": "search-bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search-bar",
                "onChange": [Function],
                "placeholder": "Search for Address",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
